### ISSUE TITLE
Infinitely Launches Instances

### ISSUE THREAD

**Issue by SteversIO** [loosebazooka/simple-spring-boot-appengine-app#2]
When i deploy this app it infinitely launches instances.  One after another...

**Comment by meltsufin** [comment]
This sample is really outdated. For one, it should use `env: flex` in app.yaml instead of `vm: true`.

**Comment by SteversIO** [comment]
I figured it was, but I'm having a heck of a time finding example code that lets me launch my Spring Boot app on App Engine.    I'm using gradle, and none of the examples i've found on the interwebs for gradle work :(. 

So i figured the only way i could reach this author is an Issue.

Do you have any suggestions?  Could I email you with what I have?

**Comment by meltsufin** [comment]
I'll try to update this sample and then you can see if it works. I suspect the app is just not responding to health checks, which causes it be restarted.

**Comment by loosebazooka** [comment]
Also, just out of curiosity, how did you get to this sample? Perhaps we need to fix some documentation somewhere that is still pointing to this.

**Comment by SteversIO** [comment]
I did a google search for App Engine, Spring Boot, Gradle.

How do you go about adding a health check response?  This documentation makes it sound like you don't have to write any code to handle health checks, and it comes out of the box.

[Here.](https://cloud.google.com/appengine/docs/flexible/java/how-instances-are-managed).

And in [this](https://cloud.google.com/appengine/docs/flexible/java/how-instances-are-managed) example repo there is no explicit code. (Maven-based, im trying to convert it to gradle so i have a working repo to go off of...)

I really truly appreciate your response.  I've spent the last day and a half trying to get my Spring Boot app working on App Engine.  I'm about to give up and just go back to K8s/docker.

**Comment by meltsufin** [comment]
@Stevers146 I've committed some changes and tried deploying. I'm not observing any application restarts. Try again and see if it's still restarting for you.

**Comment by meltsufin** [comment]
You're right this app actually responds to health checks (/_ah/health) fine. So that shouldn't be an issue. However, if you put in endpoint security in your app, the health check URL might not work.

**Comment by SteversIO** [comment]
I'll try it now thank you. (it takes a few minutes to deploy).  


This works now, not restarting.  Very odd.  

I removed all security in my own app until i could get it working unsecured.    The  _ah/health endpoint returns a 404, which the Google docs say is considered valid.

Thank you so very much.

@meltsufin  so when you say, "this app responds to health checks" do you mean vacuously, because it doesn't support the _ah/health endpoint?  Or do you mean explicitly now because you disabled health checks in the app.yaml?

**Comment by meltsufin** [comment]
When health checks are on, the App Engine infrastructure will ping /_ah/health and consider any 404 or 200 response as "healthy". It's up to you to define what "healthy" means in the context of your app and handle that endpoint. Otherwise, the default 404 is fine.

**Comment by SteversIO** [comment]
@meltsufin 
@loosebazooka 

Thank you both for being super responsive.  @meltsufin , thank you so much for your help.  Do you guys have twitter handles?

**Comment by loosebazooka** [comment]
Not really on twitter. Feel free to file bugs or issues again the gradle plugin if you encounter any https://github.com/GoogleCloudPlatform/app-gradle-plugin
